### ISSUE TITLE
Don’t catch Exception

### ISSUE THREAD

**Issue by Glutexo** [RedHatInsights/insights-core#1350]
Exceptions are not Pokémon, you don’t have to catch ’em all. There are many places in the code where a general `except` or `except Exception` is used. This universal catching shouldn’t be necessary at all, though every single case has to be refactored with caution.

[pyCodeStyle](http://pycodestyle.pycqa.org/en/latest/intro.html#error-codes) rule E722 warns about this. Even [PEP 8](https://www.python.org/dev/peps/pep-0008/) discourages of using bare _excepts_. 

Even though exception reorganization in Python 3 improved this thanks to implementing [PEP 348](https://www.python.org/dev/peps/pep-0348/#bare-except-clauses-catch-exception), it also means that the behavior is now different between the major Python versions.

I suggest gradually getting rid of the bare _excepts_ and `except Exception` cases one by one. When it looks like there is none left, remove _E722_ from Flake8 ignores and lint the code.

**Comment by csams** [comment]
There's a balance to be struck between dogmatism and pragmatism: "Practicality Beats Purity (PBP)". And there are places in core where catching top level exceptions make sense. Do you have some specific instances in mind?

**Comment by Glutexo** [comment]
I totally agree with (not only) Python’s PBP, I can however hardly imagine cases where the bare Except actually help something.

The only viable use case that comes to my mind would in the core/client’s entry point to catch any error, log it and print out an elegant “an error occured” message instead of an ugly stack trace. From what I‘ve seen recently with #1346, we‘re not even doing this.

Some examples worth improving linger in [_connection.py_](https://github.com/RedHatInsights/insights-core/blob/fef4c234c8e97ae1a8702679b2c348a14b685b50/insights/client/connection.py#L292) and [_client.py_](https://github.com/RedHatInsights/insights-core/blob/fef4c234c8e97ae1a8702679b2c348a14b685b50/insights/client/client.py):

* Like [here](https://github.com/RedHatInsights/insights-core/blob/fef4c234c8e97ae1a8702679b2c348a14b685b50/insights/client/connection.py#L292): we are catching a connection failure. If some other exception raises other that a known connection error, this is not something we want to overlook with just logging “connection failed”. A simila
* Another case might be [detecting](https://github.com/RedHatInsights/insights-core/blob/fef4c234c8e97ae1a8702679b2c348a14b685b50/insights/client/client.py#L291 invalid configuration. I understand that the config may be invalid for various reasons, but still I think the right approach would be that the specific parsers would catch their known errors and rethrow them as some known _InvalidConfigError_ instead of catching ’em all. Just as in the previous example: if something bad happens, like IO error or some code bug, it’d be swallowed and the result would be the same as if the config file is syntactically invalid.
